flow360/component/results/case_results.py

```diff
@@ -793,8 +793,7 @@ class XSlicingForceDistributionResultCSVModel(PerEntityResultCSVModel):
         CaseDownloadable.X_SLICING_FORCE_DISTRIBUTION.value, frozen=True
     )
 
-    _variables: List[str] = [_CUMULATIVE_CD_CURVE, _CD_PER_LENGTH]
-    _filter_when_zero = [_CD_PER_LENGTH]
+    _variables: List[str] = [_CUMULATIVE_CD_CURVE]
```

Comment:
why do we remove "per_length". It was used in filter when 0. Essentially it was used to get extend of data in x when excluding some boundaries. If you exclude all the boundaries you can get rid of all rows where all values are 0 and you will end up with non-zero values of which you can get x extend and zoom in plot.

Reply:
In the new algorithm, the per_length data can be calculated by the X value and cumulative_CD data, so no need to have duplication.

Reply:
I see, make sense. But now, how do I zoom in to the relevant range of data given that I don't know BBOXes of boundaries?

Reply:
updated that the CD_per_strip is calculated in the _preprocess() function of the XSlicingForceDistributionResultCSVModel(). It is added into the _filter_when_zero array too. 
It is tested in the unit tests.